src/jit/gentree.cpp

```diff
@@ -611,6 +611,60 @@ void Compiler::fgWalkAllTreesPre(fgWalkPreFn* visitor, void* pCallBackData)
 }
 
 // ------------------------------------------------------------------------------------------
+// gtCheckReg: Checks that the register has either been set, or is still set to the "no register
+// assignment" value, depending upon whether it fits any of the special cases for multi-reg call nodes.
+//
+// Arguments:
+//     compiler  -  compiler instance
+//
+// Return Value:
+//     None
+//
+// Notes:
+//     This also sets the gtRegTag to GT_REGTAG_REG.
+//
+void GenTree::gtCheckReg()
+{
+#ifdef DEBUG
+    assert((gtRegTag != GT_REGTAG_NONE) || (gtRegNum == REG_NA));
+#if CPU_LONG_USES_REGPAIR
+    assert(!"gtSetRegDefault not supported for REGPAIR");
+#endif // CPU_LONG_USES_REGPAIR
+    if (gtRegTag == GT_REGTAG_NONE)
+    {
+        gtRegTag = GT_REGTAG_REG;
+
```

Comment:
I have eliminated `gtClearReg()` as well as the `gtCheckReg()` that I added, and have simply removed the asserts that check `gtRegTag`.